bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandleAdv.java

```diff
@@ -152,8 +153,33 @@ public void asyncAddEntry(final long entryId, final byte[] data, final int offse
         asyncAddEntry(entryId, Unpooled.wrappedBuffer(data, offset, length), cb, ctx);
     }
 
+    /**
+     * Add entry asynchronously to an open ledger, using an offset and range.
+     *
+     * @param entryId
+     *            entryId of the entry to add
+     * @param data
+     *            array of bytes to be written
+     * @param offset
+     *            offset from which to take bytes from data
+     * @param length
+     *            number of bytes to take from data
+     * @param cb
+     *            object implementing callbackinterface
+     * @param ctx
+     *            some control object
+     * @throws ArrayIndexOutOfBoundsException
+     *             if offset or length is negative or offset and length sum to a
+     *             value higher than the length of data.
+     */
+    @Override
+    public void asyncAddEntry(final long entryId, final byte[] data, final int offset, final int length,
```

Comment:
is it possible to annotate this method with Evolving interface stability?

Reply:
this is old api, I don't think we have any annotation on the old api.

Reply:
Yet we do provide BC guarantees on the old api, and it's also not going away any time soon. Also, the new api isn't stable yet, so for production use people should be using the old API. For this reason we should be careful about what we add to it, and mark new/unstable stuff as such.

Reply:
yes. we do provide BC guarantees and we are carefully about any methods adding to the old api. the problem is the old api doesn't any interfaces, those are direct implementation, so we don't have any annotations on that.

Reply:
There are some interfaces annotated in the supporting classes for old api. Is it possible to use ```@Evolving``` for a single method? Not sure how to generate the javadoc anymore to check, the obvious candidates don't work.

Reply:
We only annotated some interfaces that are not gone in future releases, for example ensemble placement policy. For old api, there are no clean interfaces defined, it doesn’t make sense to spend time on annotating them since 1) they are not interfaces, 2) they are going to be deprecated in next release after the yahoo and sales force merge. That was the contract when we introduced annotations. I don’t think it is worth changing that and blocking this change at this moment.